addons/mail/static/src/model/model_manager.js

```diff
@@ -756,7 +772,10 @@ export class ModelManager {
          * considered existing.
          */
         Model.__records[record.localId] = record;
-        record._willCreate();
+        const { lifecycleHooks } = registry[Model.name];
+        if (lifecycleHooks && lifecycleHooks._willCreate) {
+            lifecycleHooks._willCreate.call(record);
```

Comment:
There is no array of hooks or possibility to patch?

Reply:
As said in the first comment of this PR:
> There is no functions yet to patch lifecycles, or to add/patch onChanges, since there is no occurrence of this use case in the current codebase.

I'm not sure we want to encourage the use of lifecycle hooks, and since this is not a actual need either, I didn't mean to change the code to support an hypothetical use case. We may never need that, and then handling lifecycle as an array would just seem odd for people reading the implementation.